docs/src/man/basics.md

```diff
@@ -0,0 +1,2002 @@
+  994 │   993     30  male        3  own      little           little          ⋯
+  995 │   994     50  male        2  own      NA               NA
+  996 │   995     31  female      1  own      little           NA
+  997 │   996     40  male        3  own      little           little
+  998 │   997     38  male        2  own      little           NA              ⋯
+  999 │   998     23  male        2  free     little           little
+ 1000 │   999     27  male        2  own      moderate         moderate
+                                                  4 columns and 985 rows omitted
+```
+
+We made a smaller *subset* of a `german` data frame to perform our operations. The `1:6` means we have taken
+the first 6 rows whereas `2:4` means we have taken the three columns form 2 to 4.
+
+```jldoctest dataframe
+julia> df1 = german[1:6, 2:4]
+6×3 DataFrame
+ Row │ Age    Sex     Job
+     │ Int64  String  Int64
+─────┼──────────────────────
+   1 │    67  male        2
+   2 │    22  female      2
+   3 │    49  male        1
+   4 │    45  male        2
+   5 │    53  male        2
+   6 │    35  male        1
+```
+
+In the following example, the operation `df1.Age = val` will be performed if `ncol(df1) == 0 || length(val) == nrow(df1)` 
+will be true. In the given condition, the logical OR operator `||` is used to perform operations on variables and values.
+The given condition will be true if either operands `ncol(df1) == 0` or `length(val) == nrow(df1)` will be true. In the case
+of `ncol(df1) == 0` will be true if both are true. Since, we are calculating number of columns using `ncol` function in our 
+data frame `df1` and the number of columns is not equal to zero so the condition is not satisfied and the result for `ncol(df) == 0`
+will be *false*. Now in the second operands which is `length(val) == nrow(df1)`, in this operand we are calculating the length of
+data frame `val` which is 6, using `length` function and calculating the number of rows present in our data frame `df1` using 
+`nrow` function. So, finally this operand tells us that the current condition is true. Result for frist operand is false and for
+second operand is true. Now, according to logical OR operator the given condition will be true.  
+
+```jldoctest dataframe
+julia> val = [80, 85, 98, 95, 78, 89]
+6-element Vector{Int64}:
+ 80
+ 85
+ 98
+ 95
+ 78
+ 89
+
+julia> ncol(df1) == 0 || length(val) == nrow(df1)
+true
+
+julia> df1.Age = val
+6-element Vector{Int64}:
+ 80
+ 85
+ 98
+ 95
+ 78
+ 89
+
+julia> df1
+6×3 DataFrame
+ Row │ Age    Sex     Job
+     │ Int64  String  Int64
+─────┼──────────────────────
+   1 │    80  male        2
+   2 │    85  female      2
+   3 │    98  male        1
+   4 │    95  male        2
+   5 │    78  male        2
+   6 │    89  male        1
+
+julia> df1.Age === val # no copy is performed
+true
+
+julia> df1[1:3, :Job] = [2, 3, 2] # set value of `:semester` in row `1:3` to values `[2, 4, 6]` *in-place*
```

Comment:
also please split code block here as you are starting a different example. In the comment please explain what in-place means.